src/snowflake/snowpark/modin/plugin/_internal/utils.py

```diff
@@ -1998,6 +1999,28 @@ def rindex(lst: list, value: int) -> int:
     return len(lst) - lst[::-1].index(value) - 1
 
 
+def error_checking_for_init(
+    index: Any, dtype: Union[str, np.dtype, ExtensionDtype]
+) -> None:
+    """
+    Common error messages for the Series and DataFrame constructors.
+
+    Parameters
+    ----------
+    index: Any
+        The index to check.
+    dtype: str, numpy.dtype, or ExtensionDtype
+        The dtype to check.
+    """
+    from modin.pandas import DataFrame
+
+    if isinstance(index, DataFrame):  # pandas raises the same error
+        raise ValueError("Index data must be 1-dimensional")
+
+    if dtype == "category":
```

Comment:
I added this check because it was not checked before - we do not support Categorical type yet. If the user passes in `dtype=category`, this makes the dtype of the Series/DataFrame `category`.

Reply:
No, it's not erroring out over here today - it's because the data itself is not categorical but should be treated like categorical if dtype is `category`.